R/g.report.part5_dictionary.R

```diff
@@ -0,0 +1,252 @@
+  getclass = function(x) {
+    if (length(x) == 0) return(NULL)
+    class = NULL
+    if (grepl("2", x)) x = unlist(strsplit(x, "2"))
+    for (i in 1:length(x)) {
+      if (x[i] == "IN") class[i] = "inactivity"
+      if (x[i] == "LIG" | x[i] == "LIPA") class[i] = "light physical activity"
+      if (x[i] == "MOD") class[i] = "moderate physical activity"
+      if (x[i] == "VIG") class[i] = "vigorous physical activity"
+      if (x[i] == "MVPA") class[i] = "moderate-to-vigorous physical activity"
+      if (x[i] == "PA") class[i] = "physical activity"
+      if (x[i] == "sleep") class[i] = "sleep"
+      if (x[i] == "wake") class[i] = "awake"
+      if (x[i] == "nonwear") class[i] = "non-wear time"
+      if (x[i] == "mostactive60min") class = "the most active 60 minutes of the day are accumulated"
+      if (x[i] == "mostactive30min") class = "the most active 30 minutes of the day are accumulated"
+    }
+    if (length(class) == 2) class = paste(class[1], "to", class[2])
+    return(class)
+  }
+  # -------------------------------------------------------------------------
+  # main script -------------------------------------------------------------
+  # -------------------------------------------------------------------------
+  # identify individual reports
+  reports = dir(file.path(metadatadir, "results"), full.names = TRUE, 
+                pattern = "^part5.*\\.csv$")
+  # read col names of each report and derive definitions
+  for (ri in 1:length(reports)) {
+    # read report
+    R = data.table::fread(reports[ri], verbose = FALSE, nrows = 2)
+    cnames = colnames(R)
+    # initialise dictionary
+    dictionary = data.frame(Variable = cnames,
+                            Definition = NA)
+    # get definitions
+    for (coli in 1:length(cnames)) {
+      # definitions will be a combination of what, window, class, unit
+      what = window = class = unit = NULL
+      # get variable names from baseDictionary
+      nam = gsub("_pla|_wei|_WD|_WE", "", cnames[coli])
+      if (grepl("Nvalid", cnames[coli])) nam = cnames[coli]
+      if (nam %in% names(baseDictionary)) {
+        what = baseDictionary[[nam]]
+      }
+      # if the variable is not in baseDictionary, extract definition:
+      if (is.null(what)) {
+        # split up variable name 
+        elements = unlist(strsplit(cnames[coli], "[.]|_"))
+        # WHAT -------------------------------------------------------------
+        if ("dur" %in% elements | "nonwear" %in% elements) {
+          what = "Time accumulated"
+        } else if ("ACC" %in% elements) {
+          what = "Mean acceleration"
+        } else if ("Nbouts" %in% elements) {
+          what = "Number of bouts"
+        } else if ("Nblocks" %in% elements) {
+          what = "Number of blocks"
+        } else if ("quantile" %in% elements) {
+          what = "Acceleration above which (percentile)"
+        } else if ("LUX" %in% elements) {
```

Comment:
Dictionary does not account for LUX segments at the moment. For example: `LUX_imputed_4_8hr_day_wei` has description "during the waking time - weighted average" and "LUX_above1000_4_8hr_day" only has "during the waking time "

So maybe easiest solution for now is to identify when this happens and replace leave those descriptions blank OR replace window == "during the waking time" by "during time segment"?

Reply:
Thanks for spotting this! I didn't see these variables because they are only calculated if `timewindow = "WW"` or `timewindow = "OO"`. Also, I could not find the documentation on the meaning of the lux variables calculated per segment. I made up some definitions after looking at the code in `g.part5.lux_persegment`, but it would be good if you double check.